src/artemis/parameters/internal_parameters.py

```diff
@@ -74,16 +75,56 @@ class InternalParameters:
     experiment_params: registry.EXPERIMENT_TYPES
 
     def __init__(self, external_params: RawParameters = RawParameters()):
```

Comment:
I don't agree, the whole job of InternalParameters init is to translate between the external and internal parameter types, that should all be in one place. Futhermore it's not plan specific, it's getting the DetectorParams from whatever the plan happens to be.

This would be much cleaner with the approach I suggested in https://github.com/DiamondLightSource/python-artemis/blob/366_simplify_params/src/artemis/parameters/internal_parameters.py where params are restricted to unique names but on the other hand can all just go in one big bucket and be easily sorted out to where they belong, but I recall you were against that - happy to entertain any other ideas?

Reply:
I've added https://github.com/DiamondLightSource/python-artemis/issues/598 from the discussion we had. Can you add more detail to it if you think it needs it?